crates/sui-core/src/authority_server.rs

```diff
@@ -414,55 +413,23 @@ impl ValidatorService {
                 &state.name,
                 certificate.clone().into(),
             );
-            let waiter = consensus_adapter.submit(transaction)?;
-            if certificate.contains_shared_object() {
-                // This is expect on tokio JoinHandle result, not SuiResult
-                waiter
-                    .await
-                    .expect("Tokio runtime failure when waiting for consensus result");
-            }
+            let _waiter = consensus_adapter.submit(transaction)?;
+            // Do not wait for the result, because the transaction might have already executed.
```

Comment:
cc @andll @lxfind this might allow for quite a bit of simplification of the consensus adapter.